Enable validation for jira issue status (#33)
ClearTax/jira-lint: README.md

 </figcaption>
</figure>

#### Issue Status Validation
Issue status is shown in the [Description](#description).

> **Why validation issue status?** 

Reviewer: ```suggestion
**Why validate issue status?** 
```

> In some cases, you may be pushing changes for a story that is set to `Done` or it may not have been pulled into working backlog. This section describes how we can help engineers avoid pushing to stories that are set to invalid statuses.

Reviewer: ```suggestion
In some cases, you may be pushing changes for a story that is set to `Done` or it may not have been pulled into working backlog. This section describes how we can help engineers avoid pushing to branches for stories that are set to statuses other than the ones allowed in the project; for example - you may want to allow PRs for stories that are in `To Do`/`Planning`/`In Progress` states.
```

The following flags can be used to validate issue status:
- `validate_issue_status`
  - If set to `true`, `jira-lint` will validate the issue status based on `allowed_issue_statuses`
- `allowed_issue_statuses`
  - This will only be used when `validate_issue_status` is `true`. This should be a comma separated list of statuses. If the detected issue's status is not in one of the `allowed_issue_statuses` then `jira-lint` will fail the status check.

**Example of invalid status**

>   <p>:broken_heart: The detected issue is not in one of the allowed statuses :broken_heart: </p>    

Reviewer: :+1: For the format of this comment :D

| `skip-branches` | A regex to ignore running `jira-lint` on certain branches, like production etc.                                                                                                                                                                                                                                    | false    | ' '     |
| `skip-comments` | A `Boolean` if set to `true` then `jira-lint` will skip adding lint comments for PR title.                                                                                                                                                                                                                         | false    | false   |
| `pr-threshold`  | An `Integer` based on which `jira-lint` will add a comment discouraging huge PRs.                                                                                                                                                                                                                                  | false    | 800     |

> | `validate_issue_status`  | An `Boolean` based on which `jira-lint` will validate the status of the detected jira issue                                                                                                                                                                                                              | false    | false   |

Reviewer: ```suggestion
| `validate_issue_status`  | A `Boolean` based on which `jira-lint` will validate the status of the detected jira issue                                                                                                                                                                                                              | false    | false   |
```

> | `allowed_issue_statuses`  | An comma separated string. The detected jira issue's status will be compared against this list and if a match is not found then the status check will fail. Requires `validate_issue_status` to be set to `true`.                                                                                        | false    | In Progress |

Reviewer: ```suggestion
| `allowed_issue_statuses`  | A comma separated list of allowed statuses. The detected jira issue's status will be compared against this list and if a match is not found then the status check will fail. *Note*: Requires `validate_issue_status` to be set to `true`.                                                                                        | false    | `"In Progress"` |
```
